source/schemas/types.txt

```diff
@@ -509,3 +509,124 @@ with the ``title`` field set to the embedded object's type name:
             }
       }
    }
+
+.. _geospatial:
+
+Geospatial Data
+---------------
+
+Geospatial data describes points and other data on the earth's surface. The 
+Realm SDKs support the creation of numerous types of geospatial data. However, 
+only persisted GeoJSON Point data can be queried. 
+
+GeoJSON Point
+~~~~~~~~~~~~~
+
+A GeoJSON Point (GeoPoint) is a single location on the Earth's surface. The GeoPoint 
+schema must contain the following fields:
+
+- ``type`` (string), which is always set to "Point"
+
+- ``coordinates`` (array of doubles). The ``coordinates`` array must contain  
+  at least 2 double values, and may contain a third:
+    
+    - The first double is the longitude of the point, between -180 and 180.
+    - The second is the latitude, between -90 and 90.
+    - The optional third value represents the elevation/altitude of the point, 
+      in meters.
+
+To store a GeoPoint in Atlas for later querying with :ref:`Atlas Device Sync <sync>`, 
+the following rules also apply:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- Additional properties are allowed within the ``coordinates`` array, as long as 
+  they are doubles.
+- Additional properties in the GeoPoint object are also allowed.
+ 
+The following is the Device Sync schema of an object with an embedded GeoPoint 
+property named "location":
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "title": "MyObject",
+     "properties": {
+       "location": {
+         "bsonType": "object",
+         "required": [ "type" ],
+         "properties": {
+           "type": {
+             "bsonType": "string",
+           },
+           "coordinates": {
+             "bsonType": "array",
+             "items": {
+               "bsonType": "double"
+             }
+           }
+          }
+         }
+       },
+    "additionalProperties": true
```

Comment:
Also wanted to ping and check if setting "additionalProperties" will lead to this property being interpreted as a dictionary versus an object. Happy to also follow up with my team if no one knows

Reply:
@jonatlee I don't know. If you don't mind checking, I'd appreciate it. How does it change the schema definition, though?

Reply:
I can check in with the team, but I guess I also don't think it's totally necessary to include here in the example. I suspect most people will just want to know how to specify a GeoJSON point without any additional properties within the point, so having the minimum definition as an example should suffice imo. Let me know if you feel strongly about this though

Reply:
Yeah, so just checked on this -- specifying "additionalProperties" in addition to "properties" for an embedded object will make the entire schema sync-incompatible. So let's omit specifying "additionalProperties" in the example